lib/licensed/sources/nuget.rb

```diff
@@ -17,10 +17,13 @@ class NuGetDependency < Licensed::Dependency
         PROJECT_URL_REGEX = /<projectUrl>\s*(.*)\s*<\/projectUrl>/ix.freeze
         PROJECT_DESC_REGEX = /<description>\s*(.*)\s*<\/description>/ix.freeze
 
-        def initialize(name:, version:, path:, search_root: nil, metadata: {}, errors: [])
-          super(name: name, version: version, path: path, search_root: search_root, metadata: metadata, errors: errors)
-          @metadata["homepage"] = project_url if project_url
-          @metadata["summary"] = description if description
+        # Returns the metadata that represents this dependency.  This metadata
+        # is written to YAML in the dependencys cached text file
+        def license_metadata
+          super.tap do |record_metadata|
+            record_metadata["homepage"] = project_url if project_url
+            record_metadata["summary"] = description if description
+          end
```

Comment:
Nice! I wanted to do something like this but I think wasn't sure if it was cool to override `license_metadata`. Also TIL about `tap`, thanks for the knowledge 👍

Reply:
Thinking of it in terms on C# is it akin to overriding virtual property? That is:
1) Get property value from base class
2) Assign two metadata values
3) Return modified property

Reply:
@paveliak yup 👍 .  In ruby all methods can be overridden by default, though there is no special syntax like `virtual` and `override` which explicitly tell which methods can be overridden and which methods are overriding base classes.  The call to `super` here is equivalent to calling `base.licensed_metadata()` in c#.